urlaf656038a0_Salihan04_MOBLIMA_tgz-pJ8-com_moblima_util_dataTestJ8/src/com/moblima/presentation/SearchListMovieAndCreateBooking_CLI.java

```diff
@@ -0,0 +1,474 @@
+				
+					printBookingPurchaseInvoice(booking);
+				}
+				break;
+		}
+	}
+	
+	public void printBookingPurchaseInvoice(Booking booking) {
+		int no_of_tickets = booking.getMovieTickets().size();
+		MovieTicket movieTicket = booking.getMovieTickets().get(0);
+		ShowTime showTime = movieTicket.getShowTime();
+		MovieGoer movieGoer = booking.getMovieGoer();
+		Transaction transaction = booking.getTransaction();
+		float ticket_price = ticketPriceBL.getPrice(showTime, movieGoer);
+		float total = no_of_tickets * ticket_price;
+		float gst = (float)((total + 1.5) * 0.07);
+		float grand_total = (float)((total + 1.5) * 1.07);
+		
+		System.out.println();
+		System.out.println("\tBOOKING PURCHASE INVOICE");
+		System.out.println("\t========================");
+		System.out.println();
+		System.out.println("Booking Ref No : " + booking.getBookingID());
+		System.out.println("TID            : " + transaction.getTID());
+		System.out.println();
+		System.out.println("Description\t\tQty\tAmount");
+		System.out.println("--------------------------------------");
+		System.out.println(movieTicket.getTicketType() + " Ticket ($" + ticket_price + ")\t" + no_of_tickets + "\t$" + total);
+		System.out.println("Online Booking Fee\t\t$" + 1.50);
+		System.out.println("\t\t\t\t------");
+		System.out.println("\t\t\tGST 7%:\t$" + gst);
+		System.out.println("\t\t\tTotal :\t$" + grand_total);	
+	}
+	
+	public void printMovieInfo(Movie movie) {
+		int index = 0;
+		_showTimes = new ArrayList<ShowTime>();
+		
+		System.out.println();
+		System.out.println(movie.getTitle());
+		System.out.println("=================================");
+		System.out.println("AVAILABLE IN\t: " + movie.getType());
+		System.out.println("CAST\t\t: " + movie.getCast());
+		System.out.println("DIRECTOR\t: " + movie.getDirector());
+		System.out.println("LANGUAGE\t: " + movie.getLanguage());
+		System.out.println("OPENING\t\t: " + movie.getOpening());
+		System.out.println("RUNTIME\t\t: " + movie.getRunTime());
+		System.out.println("RATING\t\t: " + movie.getRating());
+		System.out.println("Description\t: " + movie.getDetails());	
+		
+		// Printing movie showtimes
+		System.out.println();
+		System.out.println("SHOWTIMES");
+		System.out.println("=========");
+		
+		for(Cineplex cineplex: cineplexBL.getCineplexes()) {
+			List<Cinema> cinemas = cineplex.getCinemas(movie);
+			
+			if(cinemas == null) {
+				continue;
+			}
+			
+			System.out.println(cineplex.getCineplexName());
+			System.out.println("..................................");
+			
+			// Go through every cinemas
+			for(Cinema cinema: cinemas) {
+				List<ShowTime> showTimes = cinema.getShowTimes(movie);
+				
+				if(showTimes == null) {
+					continue;
+				}
+					
+				Hashtable<String, List<ShowTime>> showTimeHashTable = new Hashtable<String, List<ShowTime>>();
+				//Preprocessing for showTime to follow the cathay format
+				SimpleDateFormat sdf = new SimpleDateFormat("d MMM, E");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.